README.md

```diff
@@ -96,16 +96,16 @@ export default class EditorComponent extends Component {
           cellContent: 'block+',
           inlineBorderStyle: { width: '0.5px', color: '#CCD1D9' },
         }),
-        heading,
+        heading: heading(),
```

Comment:
Thinking about this, I wonder if the best approach is to, rather than change the nodespec to a function, to add a `WithConfig` function version, much in the same way as `docWithConfig`. We could then also export a constant version which is just the result of calling the function with no argument. e.g.

```
export const headingWithConfig: (config?: Config) => SayNodeSpec;
/** @deprecated Use headingWithConfig instead */
export const heading = headingWithConfig();
```

This would potentially avoid having to change any clients that don't want to use any of the new functionality. I guess unless we want it to be an explicit change that users know about?

Reply:
Oops, just re-read your comment and realised that this is essentially your 'to limit the breaking changes' comment in your summary. I guess I see that as not too much overhead as it doesn't introduce any duplication, just an extra exported function for each RDFa aware nodespec compared to what existed before.

Reply:
Yeah that's true, it would indeed not introduce too much overhead, although I would not deprecate the older 'static' version.

Reply:
I like deprecating as I'm imagining the path of someone migrating to `rdfaAware: true`.
- If there is no deprecation, they have no idea which static NodeSpecs need to be switched to a function, they need to go through the code or hope for a list in the docs.
- If we deprecate the static version, either
  - they're migrating straight to `rdfaAware: true` and they can replace the deprecated statics,
  - or they are migrating from `rdfaAware: false` and they can look at what config options are available for each function NodeSpec.

Reply:
Yeah, that's fair. I'm just not sure about the naming of the configurable node-specs.

Reply:
yes, we're indeed talking about deprecating non-function node-specs, I'm just wondering if we can provide a better API than consumers of this package having to use syntax like `block_rdfa: block_rdfa()` the whole time.

Reply:
I restored the classic node-specs, and create `WithConfig` functions where relevant. I'm now thinking of a better solution on how to configure node-specs.

Reply:
I'm not too worried about convenience in the config process, as it's already pretty terrible (on purpose)
clarity and explicitness over convenience for now, and then we add a convenience layer later on

Reply:
I deprecated all static node-specs that now have a configurable option